### ISSUE TITLE
Migration to tokio

### ISSUE THREAD

**Issue by gabomgp** [dwrensha/capnproto-rust#67]
Hello. I haven't real experience with capnproto or rust, except per two toy micro proyects. But i wish help to port capnproto to tokio and future-rs. I'll remove the dependencies on gj and gj-io, add tokio and futures-rs and fix compiler errors until the compiler dont scream. Is that good idea?

P.d. sorry by my english

**Comment by dwrensha** [comment]
I agree that we should migrate to futures-rs and tokio-core. I've made some progress in that direction with the [capnp-futures-rs](https://github.com/dwrensha/capnp-futures-rs) crate, but there is still a lot to do. Please keep me posted on any progress you make, and feel free to ask me any questions.

**Comment by gabomgp** [comment]
Well, i'm studying the code, and the rpc protocol specification, read the code of the new and old create, and i'm lost yet. And I have one question: 

1) How you visualize the migration, ¿what's the next file you could migrate?, because i don't know where to start really

**Comment by gabomgp** [comment]
Well, the light is coming to me. My attemp is starting here: https://github.com/gabomgp/capnp-rpc-tokio. There is ver little yet, but i hope i can help.

**Comment by dwrensha** [comment]
> 1. How you visualize the migration, ¿what's the next file you could migrate?

I would start by updating [this declaration](https://github.com/dwrensha/capnproto-rust/blob/8bf28185e48e2ce3a751aba79e0ca7e8eeb4f415/src/capability.rs#L30) in the `capnp` crate to be

```rust
pub type Promise<T,E> = Box<::futures::Future<Item=T,Error=E>>;
```

I would then update the `capnpc` crate so that code generation still produces code that type checks. Then I would start on `capnp-rpc`.

**Comment by dwrensha** [comment]
I've started working on this: https://github.com/dwrensha/capnproto-rust/tree/futures, https://github.com/dwrensha/capnpc-rust/tree/futures, https://github.com/dwrensha/capnp-rpc-rust/tree/futures

**Comment by salpalvv** [comment]
While I eagerly await the futures implementation, I went ahead and dove into capnproto-rust with the gj dep because I plan to use capnproto-rpc for my project. 

One issue that I had was the lack of very simple examples. I had to go to [this blog post](https://abronan.com/getting-started-with-capnproto-rpc-for-rust/) to find a simple hello world example. 

I would like to suggest that when you have the futures implementation that you have an extremely simple, contrived example (hello world is always great) so even I may follow along. Futures are very complex as is and I imagine having to follow along with a pubsub and calculator examples may prove difficult.
